### PR TITLE
Migrate to Ubuntu 22.04

### PR DESCRIPTION
**Pull request by hughcapet**
- Change Spilo base image to Ubuntu 22.04
- Add USE_OLD_LOCALES env to enable usage of locales from 18.04
- Bump Postgis version to 3.3
- Build libgdal30
- Remove deprecated apt-key
- Always build WAL-G from source

### CODE REVIEW

File: postgres-appliance/build_scripts/prepare.sh [hunk 1]
```diff
@@ -30,26 +30,18 @@ fi
 # Cleanup all locales but en_US.UTF-8 and optionally specified in ADDITIONAL_LOCALES arg
 find /usr/share/i18n/charmaps/ -type f ! -name UTF-8.gz -delete
```
Comment (CyberDem0n) [inline postgres-appliance/build_scripts/prepare.sh:31]: These lines are already in the [locales.sh](https://github.com/zalando/spilo/pull/770/files#diff-0f1e719f5a23de80bcd623f061a08c2d7726e2976c880e6b752f4c3a4b220206R13-R14) and should be removed from here.

File: postgres-appliance/build_scripts/prepare.sh [hunk 2]
```diff
@@ -30,26 +30,18 @@ fi
...
+ln -s /run/locale-archive /usr/lib/locale/locale-archive
+ln -s /usr/lib/locale/locale-archive.22 /run/locale-archive
 
 # Add PGDG repositories
 DISTRIB_CODENAME=$(sed -n 's/DISTRIB_CODENAME=//p' /etc/lsb-release)
 for t in deb deb-src; do
     echo "$t http://apt.postgresql.org/pub/repos/apt/ ${DISTRIB_CODENAME}-pgdg main" >> /etc/apt/sources.list.d/pgdg.list
 done
-curl -s -o - https://www.postgresql.org/media/keys/ACCC4CF8.asc | apt-key add -
+curl -s -o - https://www.postgresql.org/media/keys/ACCC4CF8.asc | gpg --dearmor | tee /etc/apt/trusted.gpg.d/apt.postgresql.org.gpg >/dev/null
```
Comment (CyberDem0n) [inline postgres-appliance/build_scripts/prepare.sh:44]: Why not just the following?
```suggestion
curl -s -o - https://www.postgresql.org/media/keys/ACCC4CF8.asc | gpg --dearmor > /etc/apt/trusted.gpg.d/apt.postgresql.org.gpg
```

File: postgres-appliance/scripts/wal-e-wal-fetch.sh [hunk 3]
```diff
@@ -182,9 +187,11 @@ function try_to_promote_prefetched() {
 
 echo "$$ $SEGMENT"
 
-readonly PREFETCHDIR=$(dirname "$DESTINATION")/.wal-e/prefetch
+PREFETCHDIR=$(dirname "$DESTINATION")/.wal-e/prefetch
+readonly PREFETCHDIR
 if [[ $prefetch -gt 0 && $SEGMENT =~ ^[0-9A-F]{24}$ ]]; then
-    readonly PREFETCHES=($(generate_next_segments "$prefetch"))
+    IFS=" " read -r -a PREFETCHES <<< "$(generate_next_segments "$prefetch")"
```
Comment (CyberDem0n) [inline postgres-appliance/scripts/wal-e-wal-fetch.sh:193]: it isn't doing quite the same as the original line.

File: postgres-appliance/build_scripts/prepare.sh (hunk 2 above)
Comment (hughcapet) [reply postgres-appliance/build_scripts/prepare.sh:44]: silly copy-pasting from docs...

File: postgres-appliance/scripts/wal-e-wal-fetch.sh [hunk 4]
```diff
@@ -182,9 +187,12 @@ function try_to_promote_prefetched() {
 
 echo "$$ $SEGMENT"
 
-readonly PREFETCHDIR=$(dirname "$DESTINATION")/.wal-e/prefetch
+PREFETCHDIR=$(dirname "$DESTINATION")/.wal-e/prefetch
+readonly PREFETCHDIR
 if [[ $prefetch -gt 0 && $SEGMENT =~ ^[0-9A-F]{24}$ ]]; then
-    readonly PREFETCHES=($(generate_next_segments "$prefetch"))
+    PREFETCHES=()
+    while IFS=" " read -r line; do PREFETCHES+=("$line"); done < <(generate_next_segments "$prefetch")
```
Comment (CyberDem0n) [inline postgres-appliance/scripts/wal-e-wal-fetch.sh:194]:
```suggestion
    mapfile -t PREFETCHES < <(generate_next_segments "$prefetch")
```

**Comment by CyberDem0n** [comment]
:+1:

**Comment by hughcapet** [comment]
👍

**Pull request merged**